frontend/packages/topology/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { CatalogItem } from '@console/plugin-sdk';
+import QuickSearchBar from './QuickSearchBar';
+import QuickSearchContent from './QuickSearchContent';
+import './QuickSearchButton.scss';
+import {
+  getQueryArgument,
+  removeQueryArgument,
+  setQueryArgument,
+} from '@console/internal/components/utils';
+
+interface QuickSearchModalBodyProps {
+  allCatalogItemsLoaded: boolean;
+  searchCatalog: (query: string) => CatalogItem[];
+  namespace: string;
+  closeModal: () => void;
+}
+
+const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
+  searchCatalog,
+  namespace,
+  closeModal,
+  allCatalogItemsLoaded,
+}) => {
+  const [catalogItems, setCatalogItems] = React.useState<CatalogItem[]>(null);
+  const [searchTerm, setSearchTerm] = React.useState<string>(
+    getQueryArgument('catalogSearch') || '',
+  );
+  const [selectedItemId, setSelectedItemId] = React.useState<string>('');
+  const [selectedItem, setSelectedItem] = React.useState<CatalogItem>(null);
+  const listCatalogItems = catalogItems?.slice(0, 5);
+  const ref = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    searchTerm && allCatalogItemsLoaded && setCatalogItems(searchCatalog(searchTerm));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allCatalogItemsLoaded, searchCatalog]);
+
+  React.useEffect(() => {
+    if (catalogItems && !selectedItemId) {
+      setSelectedItemId(catalogItems?.[0]?.uid);
+      setSelectedItem(catalogItems?.[0]);
+    }
+  }, [catalogItems, selectedItemId]);
+
+  const onSearch = React.useCallback(
+    (value: string) => {
+      setSearchTerm(value);
+      setCatalogItems(value ? searchCatalog(value) : null);
+      setSelectedItemId('');
+      setSelectedItem(null);
+    },
+    [searchCatalog],
+  );
+
+  const onEscapePress = React.useCallback(() => {
```

Comment:
Small nit... rename this to `onCancel`? I think tying it to the button is a coupling aspect, whereas the functionality of "escape" is really just to cancel, exit or close... so I think if we call it that, it has more flexibility and is not tied to a single usage of the action. 

Like consider if a button got added to close the modal -- could use the same function/action but wouldn't be escape.